ua/org.eclipse.ua.tests.doc/src/org/eclipse/ua/tests/doc/internal/linkchecker/LinkTest.java

```diff
@@ -101,17 +103,14 @@ public void addHits(List<SearchHit> hits, String wordsSearched) {
 		assertEquals(Collections.emptySet().toString(), linkFailures.toString());
 	}
 
+	private static final Pattern HREF = Pattern.compile("<a href=\"([^\"]+)\"");
+
 	private Set<String> checkLinks(InputStream stream, URI currentDoc, Set<URI> knownPagesURIs) throws IOException {
 		Set<String> res = new HashSet<>();
 		try (BufferedReader in = new BufferedReader(new InputStreamReader(stream))) {
-			String inputLine;
-			while ((inputLine = in.readLine()) != null) {
-				int index = 0;
-				while ((index = inputLine.indexOf("<a href=\"", index)) > 0) {
-					int closeIndex = inputLine.indexOf('"', index + "<a href=\"".length());
-					URI href = URI
-							.create(inputLine.substring(index + "<a href=\"".length(), closeIndex).replace(" ", "%20"));
-					index = closeIndex;
+			for (String inputLine; (inputLine = in.readLine()) != null;) {
+				for (Matcher matcher = HREF.matcher(inputLine); matcher.find();) {
+					URI href = URI.create(matcher.group(1).replace(" ", "%20"));
```

Comment:
Sorry @HannesWell. I misread your code during first review. We still need to trim the regex result here to deal with URIs starting/ending with blanks.

```suggestion
					URI href = URI.create(matcher.group(1).strip().replace(" ", "%20"));
```